src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1204,10 +1283,226 @@ class Simplex_tree {
     dimension_ = max_dim - dimension_;
   }
 
+  /**
+    * @brief Adds a vertex or an edge in a flag complex, as well as all
+    * simplices of its star, defined to maintain the property
+    * of the complex to be a flag complex, truncated at dimension dim_max.
+    *
+    * In term of edges in the graph, inserting edge `[u,v]` only affects
+    * the subtree rooted at @p u.
+    *
+    * For a new node with label @p v, we first do a local expansion for
+    * computing the children of this new node, and then a standard expansion
+    * for its children.
+    * Nodes with label @p v (and their subtrees) already in the tree
+    * do not get affected.
+    *
+    * Nodes with label @p u get affected only if a Node with label @p v is in their same
+    * siblings set.
+    * We then try to insert "ponctually" @p v all over the subtree rooted
+    * at `Node(u)`. Each insertion of a Node with @p v label induces a local
+    * expansion at this Node (as explained above) and a sequence of "ponctual"
+    * insertion of `Node(v)` in the subtree rooted at sibling nodes of the new node,
+    * on its left.
+    *
+    * @param[in] u,v              Vertex_handle representing the new edge
+    * @param[in] fil              Filtration value of the edge
+    * @param[in] dim_max          Maximal dimension of the expansion
+    * @param[in] added_simplices  Contains at the end all new
+    *                             simplices induced by the insertion of the edge.
+    *
+    * @pre `SimplexTreeOptions::link_nodes_by_label` must be true.
+    * @pre `Simplex_tree::Dictionary` must sort `Vertex_handles` w/ increasing natural order `<`
+    * @pre When inserting the edge `[u,v]`, the vertices @p u and @p v have to be
+    * already inserted in the simplex tree.
+    *
+    * @warning If the edges and vertices are not inserted in the order of their
+    * filtration values, the method `make_filtration_non_decreasing()` has to be
+    * called at the end of the insertions to restore the intended filtration.
+    */
+  void insert_edge_as_flag(  Vertex_handle                   u
+                           , Vertex_handle                   v
+                           , Filtration_value                fil
+                           , int                             dim_max
+                           , std::vector<Simplex_handle>&    added_simplices)
+  {
+    static_assert(Options::link_nodes_by_label, "Options::link_nodes_by_label must be true");
+
+    if (u == v) { // Are we inserting a vertex?
```

Comment:
The idea is to have a single vector for the vertices and edges to pass, so the vertices are represented with pairs of twice the same number. 
But this a point I wanted to discuss: wouldn't it also make sense to automatically add a vertex if not found (we have to do a `find` anyway)? So the user can give them explicitly if he wants, but can also have them automatically added in a flag-like way as soon as needed.

Reply:
> wouldn't it also make sense to automatically add a vertex if not found (we have to do a find anyway)?

For normal insertion, we have 2 functions: `insert_simplex` with a precondition that the faces must already be present (cheaper), and `insert_simplex_and_subfaces` which also inserts the faces. Both options also make sense for edge flag insertion, you have to see which one is most useful for zigzags (or add both).

Reply:
> The idea is to have a single vector for the vertices and edges to pass

That single vector is somewhere in zigzag I guess? I wonder if the test u==v would make more sense in the caller of insert_edge_as_flag or here (maybe here if it behaves differently from insert_simplex). Just leaving this comment so I remember to look at it later.

Reply:
For zigzag, the method is only useful to build oscillating rips filtrations. In the building process we know when we insert a vertex or an edge, so we can easily use `insert_simplex` instead, it is just a lot of copying around and testing for a single vertex (I don't know how much the compiler optimizes that). On the other hand, inserting an edge is so much heavier that the vertex insertion is nothing compared to it anyway. And it could avoid testing (v == u) for every edge. In the end, both methods are probably fine for zigzag.
So, I would say, it depends more on how we want to interface this method for potentially more general purposes outside of zigzag?

Reply:
> That single vector is somewhere in zigzag I guess?

Yes, in `oscillating_rips_iterators.h` which will be very soon in a next PR (just need to finish the doc).

Reply:
I am okay with this solution too. It really depends on the interface we want and how much we want to let to the user. If we specify a method for the vertices, for zigzag it just means that we test `u == v` before calling the method instead of inside, so it makes no difference. So, I will just follow what you think makes more sense for Gudhi.

Reply:
If this function was really meant to be used directly by users, I think I would have had a slight preference for splitting the vertex case in a different function, but
1. this function is very specific and likely won't be much used outside of zigzag
2. sharing as is done now is not horrible
so I would suggest to keep it as you have now to avoid unnecessary work.